openstf/model/objective.py

```diff
@@ -55,6 +121,11 @@ def __call__(self, trial: optuna.trial.FrozenTrial) -> float:
         Returns:
             float: Mean absolute error for this trial.
         """
+        # Check elapsed time and after n minutes end trial and stop the study
+        elapsed = datetime.utcnow() - self.start_time
+        if elapsed > timedelta(minutes=2, seconds=0):
+            trial.study.stop()
+
         # Perform data preprocessing
         train_data, validation_data, test_data = split_data_train_validation_test(
             self.input_data,
```

Comment:
Minor remark. Perhaps not part of this PR, but this seems a little odd place to check if the data is in the right format. Probably better to have a single unit which is responsible for this check and perhaps it makes more sense to check it at construction.

Reply:
Does this comment belong to this part of the code?
If so please explain a bit further.

Reply:
![image](https://user-images.githubusercontent.com/6715707/133213686-e9aa1a86-7aff-4cb7-bda0-a85949734818.png)
The comment was about the part right below. But is probably not part of this PR. I guess we can just put in on resolve for now.